### PR TITLE
Add Audience to API

### PR DESCRIPTION
**Pull request by chamilaadhi**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml [hunk 1]
```diff
@@ -7855,6 +7855,15 @@ components:
...
           example: HTTP
+        audience:
+          type: string
+          description: The audience of the API. Accepted values are PUBLIC, PRIVATE, SINGLE, PARTNERS
+          example: PUBLIC
+          enum:
+            - PUBLIC
+            - PRIVATE
+            - SINGLE
+            - PARTNERS
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml:7866]: Shall we add only the implemented one to the API?
We can keep all 4 in the internal classes.
```suggestion
            - PUBLIC
            - SINGLE
```

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/test/java/org/wso2/carbon/apimgt/persistence/utils/PersistenceHelper.java [hunk 2]
```diff
@@ -112,6 +112,7 @@ public static GenericArtifact getSampleAPIArtifact() throws GovernanceException
         artifact.setAttribute("overview_endpointUsername",null);
         artifact.setAttribute("overview_status", "PUBLISHED");
         artifact.setId("88e758b7-6924-4e9f-8882-431070b6492b");
+        artifact.setAttribute("overview_audience", "PRIVATE");
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.persistence/src/test/java/org/wso2/carbon/apimgt/persistence/utils/PersistenceHelper.java:115]: PUBLIC?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml [hunk 3]
```diff
@@ -8010,6 +8019,15 @@ components:
...
             - SSE
+        audience:
+          type: string
+          description: The audience of the API. Accepted values are PUBLIC, PRIVATE, SINGLE, PARTNERS
+          example: PUBLIC
+          enum:
+            - PUBLIC
+            - PRIVATE
+            - SINGLE
+            - PARTNERS
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml:8030]: Same as above comment

**Review by malinthaprasan — APPROVED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/test/java/org/wso2/carbon/apimgt/persistence/utils/PersistenceHelper.java (hunk 2 above)
Comment (chamilaadhi) [reply components/apimgt/org.wso2.carbon.apimgt.persistence/src/test/java/org/wso2/carbon/apimgt/persistence/utils/PersistenceHelper.java:115]: this was added for the test case

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml (hunk 1 above)
Comment (chamilaadhi) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml]: Removed

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/test/java/org/wso2/carbon/apimgt/persistence/utils/PersistenceHelper.java (hunk 2 above)
Comment (chamilaadhi) [reply components/apimgt/org.wso2.carbon.apimgt.persistence/src/test/java/org/wso2/carbon/apimgt/persistence/utils/PersistenceHelper.java]: changed

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml (hunk 3 above)
Comment (chamilaadhi) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml]: Fixed

**Pull request merged**